File: Platform.Exceptions.Tests/Ignore/EnsureAlwaysExtensions.cs
using System.Diagnostics;
using Platform.Exceptions.ExtensionRoots;

namespace Platform.Exceptions.Tests.Ignore
{
    public static class EnsureAlwaysExtensions
    {
        [Conditional("DEBUG")]
        public static void ArgumentNotNull<TArgument>(this EnsureAlwaysExtensionRoot root, TArgument argument, string argumentName)
            where TArgument : class
        {
        }
    }
}

File: cpp/Platform.Exceptions.Tests/EnsureAlwaysExtensions.cpp
namespace Platform::Exceptions::Tests::Ignore
{
    class EnsureAlwaysExtensions
    {
        public: template <typename TArgument> static void ArgumentNotNull(EnsureAlwaysExtensionRoot root, TArgument* argument, const char* argumentName)
        {
        }
    };
}
